user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleDetailsComponent.java

```diff
@@ -130,6 +145,57 @@ private void initButtonAndFieldBar() {
     buttonAndFieldBar.addClassName("button-and-search-bar");
   }
 
+  private void onRegisterButtonClicked(ClickEvent<Button> clickEvent) {
+    Experiment experiment = context.experimentId()
+        .flatMap(experimentInformationService::find)
+        .orElseThrow();
+
+    BatchRegistrationDialog2 dialog = new BatchRegistrationDialog2(
```

Comment:
There should only be one BatchRegistrationDialog 💍 
```suggestion
    BatchRegistrationDialog dialog = new BatchRegistrationDialog2(
```

Reply:
There is now.